src/psycopmlutils/utils.py

```diff
@@ -27,11 +29,46 @@ def generate_feature_colname(
         interval_days (int): Fills out "_within_{interval_days}" in the col name.
         resolve_multiple (str): Name of the resolve_multiple strategy.
         fallback (str): Values used for fallback.
+        loader_kwargs (dict, optional): Loader kwargs. Defaults to None.
 
     Returns:
         str: _description_
     """
-    return f"{prefix}_{out_col_name}_within_{interval_days}_days_{resolve_multiple}_fallback_{fallback}"
+    if isinstance(out_col_name, str):
+        col_name = f"{prefix}_{out_col_name}_within_{interval_days}_days_{resolve_multiple}_fallback_{fallback}"
+        if loader_kwargs:
+            col_name = f"{col_name}_{format_dict_for_printing(loader_kwargs)}"
```

Comment:
Maybe convert the `str` to `[str]` and then use the same logic to handle it? Reduces code duplication.